generate-hash-list.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+git clone https://github.com/minetest/minetest
+cd minetest
+git log --oneline origin/master > minetest.log
+
+declare -A clients
+clients["dragonfireclient"]="https://github.com/dragonfireclient/dragonfireclient"
+clients["TeranClient"]="https://github.com/Teeranu/TeranClient"
+clients["TeranClientCustomized"]="https://github.com/BotBoyM1/TeranClientCustomized"
+clients["waspsaliva"]="https://github.com/VoidCosmo/waspsaliva"
+clients["OtterClient"]="https://github.com/otterminetest/OtterClient"
+
+for client in "${!clients[@]}"; do
+	git remote add ${client} ${clients[${client}]}.git
+	git fetch ${client}
+	branch=$(git remote show ${client} | sed -n "/HEAD branch/s/.*: //p") # Get default branch of remote
+	git log --oneline ${client}/${branch} > ${client}.log
+
+	# Find and drop commits that exist in official minetest too
+	echo "$(sort -k3 <(awk '{print NR" "$0}' ${client}.log minetest.log minetest.log) | uniq -uf2 | sort -nk1 | sed -r 's/^[^ ]+ ([^ ]+) (.*)$/"\1", -- \2/')" > ../hashes/${client}.txt
```

Comment:
For merging I think you could before this line copy existing to temp file outside of repo dir, append new hashes (and some of the existing) then postpone `uniq -uf2 | sort -nk1 |` so that it is on next line, reads temp file and writes to final replacing it.

Reply:
Of course that could be one liner too but it would become a lot harder to read and understand so I'd recommend breaking it up a bit. I think original commands I wrote for this were already bit too much for one liner.